app/knight.py

```diff
@@ -0,0 +1,28 @@
+class Knight:
+    def __init__(self, knight: dict) -> None:
+        self.name = knight["name"]
+        self.power = knight["power"]
+        self.hp = knight["hp"]
+        self.armour = knight["armour"]
+        self.weapon = knight["weapon"]
+        self.potion = knight["potion"]
+        self.protection = 0
+
+    def battle_preparations(self) -> None:
+        for part_of_armour in self.armour:
+            self.protection += part_of_armour["protection"]
+        self.power += self.weapon["power"]
+        if self.potion is not None:
+            # I used if method because if I do it with loops like

```

Comment:
please, remove comments
if you want to describe something additionally in pull request use comment on GitHub